### PR TITLE
commonleappdracutmodules: require the biosdevname for intel only

### PR DESCRIPTION
**Pull request by pirat89**
The biosdevname rpm is built only for X86_64 architecture. Require
this one to be installed for preparation of upgrade initrd only when
the system is intel arch.

### CODE REVIEW

**Comment by pirat89** [comment]
(Untested. I am keeping the solution check just on unit tests and reviewers)

File: repos/system_upgrade/el7toel8/actors/commonleappdracutmodules/tests/test_modscan.py [hunk 1]
```diff
@@ -41,8 +43,17 @@ def test_created_modules(monkeypatch):
         assert not required_modules
 
 
-def test_required_packages():
-    assert set(modscan._REQUIRED_PACKAGES) == set(modscan._create_initram_packages().packages)
+def test_required_packages(monkeypatch):
+    class CurrentActorMocked(object):
```
Comment (bocekm) [inline repos/system_upgrade/el7toel8/actors/commonleappdracutmodules/tests/test_modscan.py]: The test fails on the other test - `test_required_packages`.
Fixed file: https://hastebin.com/gifijanimu.py
Also there's no need for using the monkeypatch context. It's necessary only in [certain cases](https://docs.pytest.org/en/latest/reference.html#_pytest.monkeypatch.MonkeyPatch.context) which is I think not this case.

File: repos/system_upgrade/el7toel8/actors/commonleappdracutmodules/libraries/modscan.py [hunk 2]
```diff
@@ -46,7 +47,10 @@ def _create_dracut_modules():
 
 
 def _create_initram_packages():
-    return RequiredUpgradeInitramPackages(packages=_REQUIRED_PACKAGES)
+    required_pkgs = _REQUIRED_PACKAGES
+    if architecture.matches_architecture(architecture.ARCH_X86_64):
+        required_pkgs.append('biosdevname')
```
Comment (bocekm) [inline repos/system_upgrade/el7toel8/actors/commonleappdracutmodules/libraries/modscan.py]: Didn't you want to remove the biosdevname from the _REQUIRED_PACKAGES?

**Comment by pirat89** [comment]
@bocekm Oh man... Yes! I will update tests first to catch that and then I will fix it.

**Comment by pirat89** [comment]
@bocekm Fixed. See the commits. Please, when you approve it, squash them into the one.

File: repos/system_upgrade/el7toel8/actors/commonleappdracutmodules/libraries/modscan.py (hunk 2 above)
Comment (pirat89) [reply repos/system_upgrade/el7toel8/actors/commonleappdracutmodules/libraries/modscan.py]: @bocekm Oh man... Yes! I will update tests first to catch that and then I will fix it.

**Comment by pirat89** [comment]
Mering. In case of error, blame me :-)

**Pull request merged**